tests/command-nested-global.earth

```diff
@@ -0,0 +1,31 @@
+VERSION --arg-scope-and-set 0.7
+
+FROM alpine:3.15
+WORKDIR /test
+
+# foo should be overridden with the value "expected" to run this test.
+# This asserts that the overriding scope impacts the global scope through nested
+# DO calls.
+ARG --global foo=default
+
+SHARED:
+    COMMAND
+    RUN test "$foo" == "expected"
```

Comment:
I initially scratched my head as to why this would pass; I think it's worthwhile explicitly pointing this out:
```suggestion
    # Note that foo has not yet been declared in the command, therefore we reference the globally declared arg
    RUN test "$foo" == "expected" 
```